src/OrchardCore/OrchardCore.ContentManagement.Display/ContentDisplay/ContentPartDisplayDriverTPart.cs

```diff
@@ -14,6 +15,39 @@ namespace OrchardCore.ContentManagement.Display.ContentDisplay
     /// <typeparam name="TPart"></typeparam>
     public abstract class ContentPartDisplayDriver<TPart> : DisplayDriverBase, IContentPartDisplayDriver where TPart : ContentPart, new()
     {
+        private ContentTypePartDefinition _typePartDefinition;
```

Comment:
I'm not sure as i'm reading this but could we have the same instance of `ContentPartDisplayDriver` used for the same part but for 2 different types? And / or for 2 different named part?

Reply:
Even if it was the case, the value would be set again on the second usage. As long as you return a Shape.